File: API/Helpers/UserParams.cs
using System;

namespace API.Helpers
{
    public class UserParams
    {
        private const int MAX_PAGE_SIZE = 50;

        public int CurrentPage { get; set; } = 1;

        private int _pageSize = 10;
        public int PageSize
        {
            get => _pageSize; 
            set
            {
                _pageSize = Math.Min(value, MAX_PAGE_SIZE);
                _pageSize = Math.Max(_pageSize, 1);
            }
        }

        public string Gender { get; set; }
        public int? MinAge { get; set; }
        public int? MaxAge { get; set; }
        public string OrderBy { get; set; } = "LastActive";
    }
}

File: client/src/app/_models/user-params.ts
import { UserToken } from './user-token';

class PaginationParams {
  currentPage: number = 1;
  pageSize: number = 5;
}

export class UserParams extends PaginationParams {
  gender: string = undefined;
  minAge: number = 18;
  maxAge: number = 99;
  orderBy = 'LastActive';

  constructor(user: UserToken) {
    super();
    this.gender = user.gender === 'male' ? 'female' : 'male';
  }
}

export class LikesParams extends PaginationParams {
  predicate = 'Liked';
}

export class MessageParams extends PaginationParams {
  container = 'Unread';
}
